newsfragments/1021.misc.rst

```diff
@@ -0,0 +1,3 @@
+Inheriting from :exc:`trio.Cancelled` exception and :class:`CancelScope` is now blocked.
```

Comment:
Wording thoughts:
* we usually use the active voice
* I think it should just be "instantiating X", not "instantiating from X"
* Cancelled being not-publicly-instantiable is not a user-visible change, since it's always been like that

Maybe something like: "Any attempt to inherit from CancelScope now raises TypeError. (Trio has never been able to safely support subclassing here; this change just makes it more obvious.)"